src/lumigo_tracer/utils.py

```diff
@@ -161,3 +162,62 @@ def is_aws_environment():
     :return: heuristically determine rather we're running on an aws environment.
     """
     return bool(os.environ.get("LAMBDA_RUNTIME_DIR"))
+
+
+class Frame:
+    MAX_VAR_LEN = 200
+
+    total_frames_size: int = 0
+
+    def __init__(self, frame_info: inspect.FrameInfo):
+        self.lineno: int = frame_info.lineno
+        self.file_name: str = frame_info.filename
+        self.function: str = frame_info.function
+        Frame.total_frames_size += len(str(self.lineno)) + len(self.file_name) + len(self.function)
```

Comment:
You will crash in a max_recursion_depth - because you always insert that information.
I would suggest changing the structure that you built, and create a single function (not class!) that returns the final dictionary that represents a frame, and decide outside the function whether to use it or not...

Reply:
As discussed, I'm passing `free_space` so the function can decide what to do.